drivers/dma/xilinx/xilinx_dma.c

```diff
@@ -653,18 +650,49 @@ static void xilinx_dma_start_transfer(struct xilinx_dma_chan *chan)
 	tail_segment = list_last_entry(&tail_desc->segments,
 				       struct xilinx_dma_tx_segment, node);
 
+	/* If channel is not halted, the tail descriptor's next_desc points to
+	 * chan->seg_reserve.  Swap head_segment and chan->seg_reserve, keeping
+	 * Buffer Descriptor contents from head_segment. */
+	{
+	    struct xilinx_dma_tx_segment *old_head, *new_head;
```

Comment:
Thank you for pointing this out. A bunch of style fixes is in my TODO list. BTW, since I'm not on it right now, patches welcome :)